File: acad-antecedenteMedico-service/src/main/java/com/acad/antecedentemedicoservice/Entity/AntecedenteMedico.java
package com.acad.antecedentemedicoservice.Entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "antecedente_medico")
public class AntecedenteMedico {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "idAntecedenteMedico")
    private Long idAntecedenteMedico;

    @Column(name = "edad", length = 45)
    private String edad;

    @Column(name = "altura", length = 45)
    private String altura;

    @Column(name = "peso", length = 45)
    private String peso;

    @Column(name = "grupoSanguineo", length = 45)
    private String grupoSanguineo;

    @Column(name = "enfermedadesPadecidas", length = 45)
    private String enfermedadesPadecidas;

    @Column(name = "antecedenteDeHospitalizacionoDeCirugias", length = 45)
    private String antecedenteDeHospitalizacionoDeCirugias;

    @Column(name = "seEncuentraMedicado", length = 45)
    private String seEncuentraMedicado;

    @Column(name = "presentaDiscapacidad", length = 45)
    private String presentaDiscapacidad;

    @Column(name = "alergias", length = 45)
    private String alergias;

    @Column(name = "estadoPsicologico", length = 45)
    private String estadoPsicologico;
}

File: acad-front/src/app/category/antecedenteMedico/antecedenteMedico.ts
import { Component, OnInit } from '@angular/core';
import { CommonModule, NgFor } from '@angular/common';
import { Router } from '@angular/router';
import { FormsModule } from '@angular/forms';

import {NewAntecedenteMedicoComponent} from './new/new-antecedente-medico.component';
import {AntecedenteMedicoModel} from '../models/antedecedente-medico-model';
import {AntecedenteMedicoService} from '../../core/service/antecedenteMedico/antecedente-medico.service';

@Component({
  selector: 'app-apoderado',
  standalone: true,
  templateUrl: './antecedenteMedico.html',
  styleUrl: './antecedenteMedico.scss',
  imports: [CommonModule,FormsModule,NewAntecedenteMedicoComponent],
})
export class AntecedenteMedico implements OnInit {



  antecedentesMedicos: AntecedenteMedicoModel[] = [];
  nuevoAntecedenteMedicoVisible = false;


  constructor(
    private antecedenteMedicoService: AntecedenteMedicoService,
    private router: Router
  ) {}



  nuevoAntecedenteMedico: AntecedenteMedicoModel = {
    edad: '',
    altura: '',
    peso: '',
    grupoSanguineo: '',
    enfermedadesPadecidas: '',
    antecedenteDeHospitalizacionoDeCirugias: '',
    seEncuentraMedicado: '',
    presentaDiscapacidad: '',
    alergias: '',
    estadoPsicologico: ''

  };

  mostrarFormularioNuevo(): void {
    this.nuevoAntecedenteMedicoVisible = !this.nuevoAntecedenteMedicoVisible;
  }

  onNuevoGuardado(): void {
    this.nuevoAntecedenteMedicoVisible = false;
    this.cargarAntecedentesMedicos();
  }




  ngOnInit(): void {
    this.cargarAntecedentesMedicos();
  }

  cargarAntecedentesMedicos(): void {
    this.antecedenteMedicoService.getAntecedenteMedico().subscribe((data) => {
      this.antecedentesMedicos = data;
    });
  }

  editar(id?: number): void {
    if (id != null) {
      this.router.navigate(['/antecedentes-medicos/editar', id]);
    }
  }

  eliminar(id?: number): void {
    if (id == null) return;

    if (confirm('¿Estás seguro de eliminar este antecedente Medico?')) {
      this.antecedenteMedicoService.delete$(id).subscribe(() => {
        this.antecedentesMedicos = this.antecedentesMedicos.filter(a => a.idAntecedenteMedico !== id);
      });
    }
  }
}
